### PR TITLE
DDCE-976: changes related to data Conversion for 'Charity' Object

### PR DESCRIPTION
**Pull request by abhijeetardale**
# DDCE-976

##Bug fix / New feature (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

**Review by robert-webb-hmrc — CHANGES_REQUESTED** [review]
Could we get this rebased please, Also these changes require an acceptance test update to the IDs on the "Where does your charity operate?" page.

File: app/models/submission/CharityTransformer.scala [hunk 1]
```diff
@@ -16,7 +16,177 @@
...
+  }
+
+  def userAnswersToCharityOrganisation : Reads[JsObject] = {
+    (
+      (__ \ 'charityRegulator ).read[JsArray].flatMap { _ =>
+        (__ \ 'charityOrganisation ).json.copyFrom(userAnswersToRegulator)
+      } orElse doNothing  and
+      (__ \ 'charityOrganisation \ 'registeredRegulator).json.copyFrom((__ \ 'isCharityRegulator).json.pick) and
+      ((__ \ 'charityOrganisation \ 'nonRegReason).json.copyFrom((__ \ 'selectWhyNoRegulator).json.pick) orElse doNothing) and
+      ((__ \ 'charityOrganisation \ 'notRegReasonOtherDescription).json.copyFrom((__ \ 'whyNotRegisteredWithCharity).json.pick) orElse doNothing)
```
Comment (anburoach) [inline app/models/submission/CharityTransformer.scala:60]: Replace 'notRegReasonOtherDescription with 'otherReason

File: test/models/submission/CharityTransformerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,907 @@
...
+          _.set(SelectWhyNoRegulatorPage, SelectWhyNoRegulator.Other).flatMap(
+          _.set(WhyNotRegisteredWithCharityPage, "notRegisteredReason"))
+        ).success.value
+
+        val expectedJson =
+          """{
+            |"charityOrganisation": {
+            |			"registeredRegulator": false,
+            |			"nonRegReason": "7",
+            |			"notRegReasonOtherDescription": "notRegisteredReason"
```
Comment (anburoach) [inline test/models/submission/CharityTransformerSpec.scala:230]: replace 'notRegReasonOtherDescription with 'otherReason

**Review by anburoach — CHANGES_REQUESTED** [review]
Changes requested

**Review by anburoach — APPROVED** [review]

**Pull request merged**